tests/e2e/e2e_util_test.go

```diff
@@ -978,3 +978,78 @@ func (s *IntegrationTestSuite) executeRedelegate(c *chain, valIdx int, endpoint
 	s.T().Logf("%s successfully redelegated %s from %s to %s", delegatorAddr, amount, originalValOperAddress, newValOperAddress)
 }
+
+func (s *IntegrationTestSuite) execSetWithrawAddress(c *chain, valIdx int, endpoint, fees string, delegatorAddress string, newWithdrawalAddress string, homePath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("Setting distribution withdrawal address on chain %s for %s to %s", c.id, delegatorAddress, newWithdrawalAddress)
+	gaiaCommand := []string{
+		"gaiad",
+		"tx",
+		"distribution",
+		"set-withdraw-addr",
+		newWithdrawalAddress,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, delegatorAddress),
+		fmt.Sprintf("--%s=%s", flags.FlagGasPrices, fees),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, c.id),
+		fmt.Sprintf("--%s=%s", flags.FlagHome, homePath),
+		"--keyring-backend=test",
+		"--output=json",
+		"-y",
+	}
+
+	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
+	s.T().Logf("Successfully set new distribution withdrawal address for %s to %s", delegatorAddress, newWithdrawalAddress)
+}
+
+func (s *IntegrationTestSuite) execWithdrawReward(c *chain, valIdx int, endpoint, fees string, delegatorAddress string, validatorAddress string, homePath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("Withdrawing distribution rewards on chain %s for delegator %s from %s validator", c.id, delegatorAddress, validatorAddress)
+	gaiaCommand := []string{
+		"gaiad",
+		"tx",
+		"distribution",
+		"withdraw-rewards",
+		validatorAddress,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, delegatorAddress),
+		fmt.Sprintf("--%s=%s", flags.FlagGasPrices, "300uatom"),
+		fmt.Sprintf("--%s=%s", flags.FlagGas, "auto"),
+		fmt.Sprintf("--%s=%s", flags.FlagGasAdjustment, "1.5"),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, c.id),
+		fmt.Sprintf("--%s=%s", flags.FlagHome, homePath),
+		"--keyring-backend=test",
+		"--output=json",
+		"-y",
+	}
+
+	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
+	s.T().Logf("Successfully withdrew distribution rewards for delegator %s from validator %s", delegatorAddress, validatorAddress)
+}
+
+func (s *IntegrationTestSuite) queryGaiaTx(endpoint, txHash string) error {
+	resp, err := http.Get(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs/%s", endpoint, txHash))
+	if err != nil {
+		return fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("tx query returned non-200 status: %d", resp.StatusCode)
+	}
+
+	var result map[string]interface{}
+	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	txResp := result["tx_response"].(map[string]interface{})
+	if v := txResp["code"]; v.(float64) != 0 {
+		return fmt.Errorf("tx %s failed with status code %v", txHash, v)
+	}
```

Comment:
It would be better if we could wrap it into a struct to avoid unmarshalling into a map and test all values. But it's not the case. We should assert the values inside the map:

```suggestion
	txResp, ok := result["tx_response"].(map[string]interface{})
	if !ok {
		return fmt.Errorf("invalid tx response: %s", result)
	}
	v, ok := txResp["code"]
	if !ok {
		return fmt.Errorf("invalid tx response: %s", txResp)
	}
	code, ok := v.(float64)
	if ok && code != 0 {
		return fmt.Errorf("tx %s failed with status code %v", txHash, v)
	}
```

Reply:
@Pantani per our discussion, deleted this as it's a helper function and will address the original in #1608